resource/dku-helpers.js

```diff
@@ -0,0 +1,93 @@
+/*
+Helper function to query webapp backend with a default implementation for error handling
+v 1.0.1
+*/
+
+function checkMandatoryParameters(param, webAppConfig) {
+    if (param.mandatory) {
+        var val = webAppConfig[param.name];
+        if (val == undefined || val == "") {
+            throw new Error("Mandatory column '" + param.name + "' not specified.");
+        }
+    }
+}
+
+function checkWebAppParameters(webAppConfig, webAppDesc) {
+    if (webAppDesc.topBarParams) {
+        webAppDesc.topBarParams.forEach(p => {checkMandatoryParameters(p, webAppConfig)});
+    }
+    if (webAppDesc.leftBarParams) {
+        webAppDesc.leftBarParams.forEach(p => {checkMandatoryParameters(p, webAppConfig)});
+    }
+};
+
+function checkWebAppConfig(webAppConfig) {
+    if (webAppConfig['source'] == webAppConfig['target']) {
+        throw Error("Columns must be different")
+    }
+}
+
+function quadraticScalingFunction(min, max, total, value) {
+    if (max === min) {
+        return 0;
+    } else {
+        var scale = 1 / (max - min);
+        return Math.pow(Math.max(0,(value - min)*scale), 2);
+    }
+}
+
+function isEqual(object1, object2) {
+    return JSON.stringify(object1) == JSON.stringify(object2)
+}
+
+
+dataiku.webappBackend = (function() {
+    function getUrl(path) {
+        return dataiku.getWebAppBackendUrl(path);
+    }
+
+    function dkuDisplayError(error) {
+        console.warn("backend error: ", error);
+    }
+
+    function get(path, args={}, displayErrors=true) {
+        return fetch(getUrl(path) + '?' + $.param(args), {
```

Comment:
```suggestion
        return fetch(`${getUrl(path)}?${$.param(args)}`, {
```
nitpick: I think that this syntax is most of the time clearer